### PR TITLE
[IMPROVEMENT] NFT Auto Detection Settings

### PR DESCRIPTION
**Pull request by sethkfman**
**Description**

Implement system level option to enable NFT auto detection polling and OpenSea APIs. 

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

File: app/components/Views/Settings/SecuritySettings/index.js [hunk 1]
```diff
@@ -443,6 +459,20 @@ class Settings extends PureComponent {
 		this.props.setThirdPartyApiMode(value);
 	};
 
+	toggleOpenSeaApi = (value) => {
+		const { PreferencesController } = Engine.context;
+		if (value) PreferencesController.setOpenSeaEnabled(value);
+		else {
+			PreferencesController.setOpenSeaEnabled(value);
```
Comment (blackdevelopa) [inline app/components/Views/Settings/SecuritySettings/index.js:466]: If the condition checks for a value in line 464, what value is present in line 466?

Comment (sethkfman) [reply app/components/Views/Settings/SecuritySettings/index.js:466]: value would = false/undefined it is a boolean value

File: app/core/Engine.js [hunk 2]
```diff
@@ -71,8 +71,8 @@ class Engine {
 						initialState?.PreferencesController?.useStaticTokenList === undefined ||
 						initialState.PreferencesController.useStaticTokenList,
 					// TODO: Use previous value when preferences UI is available
-					useCollectibleDetection: true,
-					openSeaEnabled: true,
+					useCollectibleDetection: false,
+					openSeaEnabled: false,
```
Comment (Cal-L) [inline app/core/Engine.js:75]: Let's create a redux migration and set these to false by default for safe measure.

File: bitrise.yml [hunk 3]
```diff
@@ -326,10 +326,10 @@ app:
           PROJECT_LOCATION_IOS: ios
         - opts:
               is_expand: false
-          VERSION_NAME: 4.0.0
+          VERSION_NAME: 4.0.1
         - opts:
               is_expand: false
-          VERSION_NUMBER: 810
+          VERSION_NUMBER: 811
```
Comment (Cal-L) [inline bitrise.yml:332]: Let's just update build version in here to 813

File: app/constants/storage.js [hunk 4]
```diff
@@ -37,6 +37,7 @@ export const LAST_APP_VERSION = `${prefix}LastAppVersion`;
 export const CURRENT_APP_VERSION = `${prefix}CurrentAppVersion`;
 
 export const WHATS_NEW_APP_VERSION_SEEN = `${prefix}WhatsNewAppVersionSeen`;
+export const NFT_DISMISS_INFO = `nftDismissInfo`;
```
Comment (Cal-L) [inline app/constants/storage.js:40]: Remove unused const

File: app/components/Views/AddAsset/index.js [hunk 5]
```diff
@@ -76,15 +87,27 @@ class AddAsset extends PureComponent {
...
 				params: { assetType, collectibleContract },
 			},
 			navigation,
+			chainId,
 		} = this.props;
 		return (
 			<SafeAreaView style={styles.wrapper} testID={`add-${assetType}-screen`}>
+				{isMainNet(chainId) && (
+					<View style={styles.infoWrapper}>
+						{!this.state.dismissNftInfo && !this.props.useCollectibleDetection && (
```
Comment (Cal-L) [inline app/components/Views/AddAsset/index.js:106]: Let's move this logic with the mainnet check

File: app/core/Engine.js (hunk 2 above)
Comment (sethkfman) [reply app/core/Engine.js:75]: Done

File: app/components/Views/Settings/SecuritySettings/index.js (hunk 1 above)
Comment (sethkfman) [reply app/components/Views/Settings/SecuritySettings/index.js]: Updated

File: app/constants/storage.js (hunk 4 above)
Comment (sethkfman) [reply app/constants/storage.js]: removed

File: app/components/Views/AddAsset/index.js (hunk 5 above)
Comment (sethkfman) [reply app/components/Views/AddAsset/index.js]: updated

File: bitrise.yml (hunk 3 above)
Comment (sethkfman) [reply bitrise.yml:332]: will do this in the cherry pick branch

File: app/core/Engine.js (hunk 2 above)
Comment (wachunei) [reply app/core/Engine.js:75]: should this be `initialState?.PreferencesController?.useCollectibleDetection ?? false` (same for `openSeaEnabled`) ?

**Pull request closed without merging**